src/lib/assignment.ts

```diff
@@ -145,14 +181,103 @@ async function reducePublish(courseId: string, srcDir: string, yamlDir: string,
   }
 }
 
+export async function getSettings(courseId: string, assignmentId:string): Promise<AssignmentSettings> {
+  return updateSettings(courseId, assignmentId, {})
+}
+
+function convertDateToLocal(date: string | undefined): Date | null {
+  if (!date) {
+    return null
+  }
+  
+ return new Date(date)
+}
+
+function toRawSettings(settings: AssignmentSettings): AssignmentSettingsRaw {
+  const res = {} as AssignmentSettingsRaw
+  if (settings.enableResetAssignmentByStudent !== undefined) {
+    res.enableResetAssignmentByStudent = settings.enableResetAssignmentByStudent
+  }
+  if (settings.disableDownloadByStudent !== undefined) {
+    res.disableDownloadByStudent = settings.disableDownloadByStudent
+  }
+  if (settings.visibilityOnDisabled !== undefined) {
+    res.visibilityOnDisabled = settings.visibilityOnDisabled
+  }
+  if (settings.visibilityOnCompleted !== undefined) {
+    res.visibilityOnCompleted = settings.visibilityOnCompleted
+  }
+  if (settings.startTime !== undefined) {
+    res.startTime = settings.startTime ? settings.startTime.toISOString() : ''
+  }
+  if (settings.endTime !== undefined) {
+    res.endTime = settings.endTime ? settings.endTime.toISOString() : ''
+  }
+  
+  if (settings.action !== undefined) {
+    res.action = settings.action
+  }
+  if (settings.penalties !== undefined) {
+    res.penalties = _.map(settings.penalties, _ => {
+      return {
+        id: _.id,
+        datetime: _.datetime.toISOString(),
+        percent: _.percent,
+        message: _.message,
+      }
+    })
+  }
+  return res
+}
+
+export async function updateSettings(courseId: string, assignmentId: string, settings: AssignmentSettings): Promise<AssignmentSettings> {
+  if (!config) {
+    throw new Error('No Config')
+  }
+  try {
+    const token = config.getToken()
+    const authHeaders = {'Authorization': `Bearer ${token}`}
+    
+    const api = bent(getApiV1Url(), 'POST', 'json', 200)
+
+    const res = await api(`/courses/${courseId}/assignments/${assignmentId}/settings`,
+        toRawSettings(settings), authHeaders) as AssignmentSettingsRaw
+    return {
+      enableResetAssignmentByStudent: res.enableResetAssignmentByStudent,
+      disableDownloadByStudent: res.disableDownloadByStudent,
+      visibilityOnDisabled: res.visibilityOnDisabled,
+      visibilityOnCompleted: res.visibilityOnCompleted,
+      startTime: convertDateToLocal(res.startTime),
+      endTime: convertDateToLocal(res.endTime),
+      action: res.action,
+      penalties: res.penalties? _.map(res.penalties, _ => {
+        return {
+          id: _.id,
+          datetime: new Date(_.datetime),
```

Comment:
Different approach new Date() or _.toIsoString

Reply:
AssignmentSettings uses `Date`, API returns and works with `string`